### PR TITLE
Decimal support

### PR DESCRIPTION
**Pull request by fm-andres-rangel**

### CODE REVIEW

**Review by fm-fede-ravchina — APPROVED** [review]

File: include/fmc++/decimal128.hpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  }
+  static fmc::decimal128 infinity() noexcept {
+    return fmc::decimal128::upcast(fmc_decimal128_inf());
+  }
+  static fmc::decimal128 quiet_NaN() noexcept {
+    return fmc::decimal128::upcast(fmc_decimal128_qnan());
+  }
+  static fmc::decimal128 signaling_NaN() noexcept {
+    return fmc::decimal128::upcast(fmc_decimal128_snan());
+  }
```
Comment (fm-fede-ravchina) [inline include/fmc++/decimal128.hpp:121]: epsilon is missing

**Review by fm-fede-ravchina — CHANGES_REQUESTED** [review]

File: include/fmc++/decimal128.hpp [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+#include <cmath>
+#include <limits>
+#include <ostream>
+
+namespace fmc {
+
+class decimal128 : public fmc_decimal128_t {
+public:
+  decimal128(const fmc_decimal128_t &a) : fmc_decimal128_t(a) {}
```
Comment (fm-fede-ravchina) [inline include/fmc++/decimal128.hpp:32]: we need a default constructor

**Review by fm-fede-ravchina — CHANGES_REQUESTED** [review]

File: include/fmc++/decimal128.hpp (hunk 2 above)
Comment (fm-andres-rangel) [reply include/fmc++/decimal128.hpp:32]: Resolved with ba89753

File: include/fmc++/decimal128.hpp (hunk 1 above)
Comment (fm-andres-rangel) [reply include/fmc++/decimal128.hpp:189]: resolved with 6cb7489

**Review by maxim-trokhimtchouk — DISMISSED** [review]

**Review by fm-ivang — APPROVED** [review]

**Review by maxim-trokhimtchouk — APPROVED** [review]

**Pull request merged**